components/Modal/index.tsx

```diff
@@ -0,0 +1,227 @@
+
+import Overlay from './Overlay';
+
+const DefaultFooter = ({
+  actionButtonText = 'Apply',
+  actionDisabled = false,
+  cancelText = 'Cancel',
+  onActionClick,
+  onCancelClick,
+}: {
+  actionButtonText?: ReactNode;
+  actionDisabled?: boolean;
+  cancelText?: ReactNode;
+  onActionClick?: (event: SyntheticEvent<HTMLButtonElement, Event>) => any;
+  onCancelClick?: (event: SyntheticEvent<HTMLButtonElement, Event>) => any;
+}) => (
+  <ButtonContainer>
+    {onCancelClick && <Button onClick={onCancelClick}>{cancelText}</Button>}
+
+    {onActionClick && (
+      <Button
+        css={css`
+          margin-left: 10px;
+        `}
+        disabled={actionDisabled}
+        onClick={onActionClick}
+      >
+        {actionButtonText}
+      </Button>
+    )}
+  </ButtonContainer>
+);
+
+const ButtonContainer = styled('div')`
+  display: flex;
+  flex-direction: row;
+`;
+
+const FocusWrapper = styled('button')`
+  align-items: center;
+  background: none;
+  border: none;
+  box-shadow: 0px 0px 4px 0px ${({ theme }: DefaultThemeObject) => theme.colors.secondary_1};
+  cursor: pointer;
+  display: flex;
+  height: 1rem;
+  justify-content: center;
+  outline: none;
+  padding: 0px;
+  transition: box-shadow 0.1s ease-in;
+  width: 1rem;
+
+  &:focus,
+  &:hover {
+    box-shadow: 0px 0px 4px 0px ${({ theme }: DefaultThemeObject) => theme.colors.grey_5};
+  }
+`;
+
+const ModalBody = styled('div')`
+  box-sizing: border-box;
+  display: flex;
+  overflow-y: auto;
+  padding: 20px;
+  width: 100%;
+`;
+
+const ModalCloseButton = ({
+  className,
+  onClick,
+}: {
+  className?: string;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+}) => {
+  const theme: DefaultTheme = useTheme();
+
+  return (
+    <FocusWrapper className={className} onClick={onClick}>
+      <Dismiss fill={theme.colors.grey_600} size={'10px'} />
+    </FocusWrapper>
+  );
+};
+
+export const ModalContainer = styled('div')`
+  ${({ theme }) => theme.typography.regular}
+  align-items: space-between;
+  background-color: ${({ theme }: DefaultThemeObject) => theme.colors.white};
+  border-radius: 3px;
+  box-shadow: 0 8px 21px 0 rgba(0, 0, 0, 0.1), 0 6px 12px 0 rgba(0, 0, 0, 0.1);
+  color: ${({ theme }: DefaultThemeObject) => theme.colors.grey_800};
+  display: flex;
+  flex-direction: column;
+  margin: 0 1rem;
+  max-height: 95vh;
+  max-width: 776px;
+  position: relative;
+  width: auto;
+`;
+
+const ModalFooter = styled('div')`
+  border-top: 1px solid ${({ theme }: DefaultThemeObject) => theme.colors.grey_4};
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-end;
+  padding: 10px 15px;
+`;
+
+const ModalTitle = styled('div')`
+  align-items: center;
+  border-bottom: 1px solid ${({ theme }: DefaultThemeObject) => theme.colors.grey_4};
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 10px 15px;
+
+  h1 {
+    ${({ theme }) => theme.typography.heading}
+    margin: 0;
+  }
+`;
+
+const ModalComponent: FC<{
```

Comment:
Then the generic modal itself (derived from UiKit's, but vastly simplified and with some improvements), which you call elsewhere, and customise the flow-specific controller logic there rather than here.